tools/docker/22.04/Dockerfile

```diff
@@ -0,0 +1,176 @@
+                       libnuma-dev \
+                       libssl-dev \
+                       wget \
+                       flex \
+                       bison \
+                       iwyu \
+                       && \
+    apt-get clean && apt-get autoclean
+
+
+# Install kotekan python dependencies
+RUN python3 -m pip install --upgrade pip && \
+    python3 -m pip install --upgrade distro-info && \
+    python3 -m pip install --upgrade --force-reinstall setuptools && \
+    python3 -m pip install --upgrade wheel && \
+    python3 -m pip install --no-cache-dir numpy && \
+    python3 -m pip install --no-cache-dir pkgconfig && \
+    python3 -m pip install --no-cache-dir --upgrade cython && \
+    python3 -m pip install --no-cache-dir click==8.0.2 && \
+    python3 -m pip install --no-cache-dir black==19.10b0 && \
+    python3 -m pip install --no-cache-dir cmake_format
+
+RUN python3 -m pip install h5py \
+                           hdf5plugin \
+                           bitshuffle
+RUN python3 -m pip install msgpack \
+                              future \
+                              requests \
+                              pyyaml \
+                              tabulate \
+                              pytest \
+                              pytest-xdist \
+                              pytest-cpp \
+                              pytest-localserver \
+                              pytest-timeout \
+                              posix_ipc
+RUN python3 -m pip install --ignore-installed flask
+
+
+# Install comet, mysql, redis for comet tests
+RUN apt-get update && \
+    apt-get install -y mysql-client \
+        libmysqlclient-dev && \
+    apt-get clean && apt-get autoclean
+RUN python3 -m pip install git+https://github.com/chime-experiment/comet.git
+RUN apt-get update && \
+    apt-get install -y redis && \
+    apt-get clean && apt-get autoclean
+
+
+# CUDA / GPU tools
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.0-1_all.deb
+RUN dpkg -i cuda-keyring_1.0-1_all.deb
+RUN apt -y update
+RUN apt install -y --no-install-recommends cuda && apt clean
+
+ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/cuda/bin
+
+RUN git clone https://github.com/kmsmith137/gputils.git \
+    && cd gputils \
+    && make install
+
+RUN git clone https://github.com/kmsmith137/n2k.git \
+  && cd n2k \
+  && make install
+
+RUN apt install -y --no-install-recommends \
+  libopencl-clang-dev opencl-headers ocl-icd-opencl-dev
+
+
+# Install OpenBLAS and clone Blaze for the eigenvalue processes
+RUN apt-get update && \
+    apt-get -y install libopenblas-dev \
+                       liblapack-dev \
+                       liblapacke-dev \
+                       && \
+    apt-get clean && apt-get autoclean && \
+    git clone https://bitbucket.org/blaze-lib/blaze.git blaze && \
+    cd blaze && git checkout v3.8 && cmake -DCMAKE_INSTALL_PREFIX=/usr/local/ && \
```

Comment:
Why does blaze change from 3.4 to 3.8?   I think we want to keep it at 3.4 since the version change caused us some problems in the past with a change to the convergence algorithm making its timing less deterministic. At the moment only one system uses blaze for a very specific case.  So it might be better to keep the version the same between Ubuntu versions as well.  Unless of course 3.4 just won't build on 22.04?

Reply:
3.4 seems to build as does 3.8, but we can stick with the old version!